src/Raven.Client/Documents/Conventions/DocumentConventions.cs

```diff
@@ -837,6 +848,51 @@ public bool TryConvertValueForQuery(string fieldName, object value, bool forRang
             return false;
         }
 
+        private void RegisterQueryValueConverter<T>(TryConvertValueToObjectForQueryDelegate<T> converter)
+        {
+            AssertNotFrozen();
+
+            int index;
+            for (index = 0; index < _listOfQueryValueToObjectConverters.Count; index++)
+            {
+                var entry = _listOfQueryValueToObjectConverters[index];
+                if (entry.Type.IsAssignableFrom(typeof(T)))
+                    break;
+            }
+
+            _listOfQueryValueToObjectConverters.Insert(index, (typeof(T), Actual));
+
+            bool Actual(string name, object value, bool forRange, out object objValue)
+            {
+                if (value is T)
+                    return converter(name, (T)value, forRange, out objValue);
+                objValue = null;
+                return false;
+            }
+        }
+
+        public void RegisterQueryValueConverter<T>(TryConvertValueToObjectForQueryDelegate<T> converter, RangeType rangeType)
+        {
+            RegisterQueryValueConverter(converter);
+
+            if (_customRangeTypes.ContainsKey(typeof(T)) == false)
+                _customRangeTypes.Add(typeof(T), rangeType);
+        }
+
+        internal bool TryConvertValueToObjectForQuery(string fieldName, object value, bool forRange, out object objValue)
+        {
+            foreach (var queryValueConverter in _listOfQueryValueToObjectConverters)
+            {
+                if (queryValueConverter.Type.IsInstanceOfType(value) == false)
```

Comment:
Just call them until you find someone that can process it, no need for the checks

Reply:
I can change it, but it was copied from the TryConvertValueForQuery version.
Shall I change both implementations?

Reply:
It seems those checks are here for fast shortcutting ...